### ISSUE TITLE
remove identities file and its git history

### ISSUE THREAD

**Issue by GeorgLink** [chaoss/grimoirelab#239]
#237 raised an important point about not providing an identities file.

I'm sure @sanacl has more insights about the legal problem with sharing personal identifiable information. 

The argument that the information already existed in the repo is not substantive. The data, in my understanding, should never have been in the repo.

Further, providing an identities file comes with the expectation that we keep it up-to-date. I don't think that we should take on that burdon.

I think we have a case for: https://help.github.com/en/github/authenticating-to-github/removing-sensitive-data-from-a-repository

**Comment by valeriocos** [comment]
To give more context to the people that want to join the discussion, we are talking about two identities.yml files
- One is in the docker [folder](https://github.com/chaoss/grimoirelab/blob/master/docker/identities.yaml) since `Jan 31, 2018`
- Another one is in the default-grimoirelab-settings [folder](https://github.com/chaoss/grimoirelab/blob/master/default-grimoirelab-settings/identities.yml) since `14 Nov 2019`. This file was added to make the docker-compose run smoothly.

> The argument that the information already existed in the repo is not substantive. The data, in my understanding, should never have been in the repo.

Since the information is already in the repo, duplicating it in the same repo doesn't change anything. This is the reason why the PR was accepted. A different story is about which target repo should be used to showcase GrimoireLab (maybe we could use a fake repo with fake users).

In the same PR, if you read carefully, you will find that:
```
Since the file already exists, the discussion about how deal with these emails addresses is probably out of the scope of this PR.
```

The discussion isn't just about to these identities files, but it concerns also the email addresses (i) visible in all file headers within the GrimoireLab repos and (ii) buried in the Git log history. Under this perspective, I'm not sure about your proposal to undoing changes and rewriting the Git history.

> Further, providing an identities file comes with the expectation that we keep it up-to-date. I don't think that we should take on that burdon.

That isn't actually true. It would be nice to have the file up-to-date (the same applies also for the other configuration files), but it's not an expectation. Remember that the idea is to have a quick way to play with GrimoireLab.

> I think we have a case for: https://help.github.com/en/github/authenticating-to-github/removing-sensitive-data-from-a-repository

Probably it's better to have a look at the GitHub legal terms and understand if we are doing something wrong.

**Comment by jgbarah** [comment]
To extend the context:
* The data is needed to have a sensible output of the grimoirelab/installed container, which produces a visualization of CHAOSS projects. If we don't have this file, we cannot produce a sensible affiliation information.
* Emails are readily available from GitHub itself if you just clone the repo and run "git log" (I'm not sure all of it, but certainly most of it)
* Affiliation in that file is only for CHAOSS contributors. If needed, we could ask for specific consent, I think.
* This is standard practice in other FOSS projects, and is needed for transparency of who is contributing to the project. See for example a [similar file in Stackalitics, for OpenStack](https://raw.githubusercontent.com/stackalytics/default_data/master/default_data.json)

However, if you still think this is a problem from the privacy point of view, i have nothing else to say ;-)

**Comment by sanacl** [comment]
Hi folks,
in terms or regulation this is what Github says:

> We do not intentionally collect User Personal Information that is stored in your repositories or other free-form content inputs. Information in your repositories belongs to you, and you are responsible for it, as well as for making sure that your content complies with our Terms of Service. Any personal information within a user's repository is the responsibility of the repository owner.

So, we (as a community) are the [data controllers](https://www.gdpreu.org/the-regulation/key-concepts/data-controllers-and-processors/). This places us in the position of letting the people know we are using their information to allow our CHAOSS demo to run smoothly. It is important that we as a community provide them a way to request to remove their data. This could be done for the next versions of the repo or in the worst case scenario by removing the file from all the versions published.

We have two options here: if all the contributors are subscribed to a project mailing list, we can send there a generic message. If they aren't, we should send them a direct message.

Last but not least, we should think about what to do with newcomers. Having their ACK to share this kind of information would be the ideal scenario.

Any comments?

**Comment by valeriocos** [comment]
Thank you @sanacl for the details and proposals. 

> We have two options here: if all the contributors are subscribed to a project mailing list, we can send there a generic message. If they aren't, we should send them a direct message.
> 
> Last but not least, we should think about what to do with newcomers. Having their ACK to share this kind of information would be the ideal scenario.

I vote for a message via the mailing list, I guess it should be enough (given also the last point mentioned by @jgbarah). WRT new comers, we can add a section in the README of this repo saying that contributor's information may appear in the CHAOSS dashboard (and if they are not happy with that we can remove their data).

WDYT?

**Comment by GeorgLink** [comment]
Thank you for the clarifications and discussion.

Because the file only contains contributors to the chaoss/grimoirelab repo and we do not claim to maintain a larger context of identities, I am okay with the proposed course of action: 
- inform via email list 
- add a disclaimer on README
- add a disclaimer on CONTRIBUTING (I added this because it's the place for contributors to check how to contribute and what it means to them)

**Comment by jsmanrique** [comment]
IMHO, GrimoireLab starting point should not require any `identities.yml` file, specially **without any clear explanation of what that file is for**, so I would vote for, either:

* remove or empty `identities_file = [/home/bitergia/conf/identities.yml]` line from any `setup.cfg` deployment file in this repository. Of course, I would remove/delete any `identities.yml` file in this repository.

or

* remove any content in any `identities.yml` file in this repository.

Regarding CHAOSS community "identities" data, AFAIK, they are part of the [configurations files](https://gitlab.com/Bitergia/c/CHAOSS/profiles) used to build [the official CHAOSS GrimoireLab Dashboard](https://chaoss.biterg.io). I've thought that CHAOSS community has been informed about that.

**Comment by jgbarah** [comment]
@jsmanrique can you pleas elaborate on your rationale for removing `identities.yml`, assuming the community is properly informed?

**Comment by jsmanrique** [comment]
1. The identities file here, AFAIK, is not the one being used to build chaoss.biterg.io
2. The identities file is not needed to start playing with GrimoireLab, so it only adds an additional step to be aware of
3. The identities file creates conflicts with data changed with HatStall (another issue that need to be addressed AFAIK)
4. The profiles described in the identities file provided in this repo is related with CHAOSS project, so it's useless for analyzing any other project

So, I would rephrase your question to: Could you please elaborate the rationale for having an `identities.yml` file to start playing with GrimoireLab?

**Comment by jgbarah** [comment]
1. That file is for use with the grimoirelab/full container, which is intended as a demonstrator. As such, it builds a dashboard for CHAOSS. No input nor configuration needed, except for your GitHub token. It could be used for chaoss.biterg.io, and in general, I try to maintain them in sync (I'm not sure if they are exactly the same now, but they should).

2. I agree, but it is also a nice thing to have. If you use the container without providing a customized file, it will still work (it will only not recognize these identities, since they are not likely to work in your project of choice).

3. Hatstall is still not integrated in the `grimoirelab/full` container. And in any case, it causes no conflict: it is just a starting point, that you can later change with hatstall if you want (AFAIK).

4. Of course. Because it is intended to produce a nice dashboard when the container is run, out of the box, for CHAOSS (as a demonstrator). Documentation explains how to change it (and if documentation is not clear enough, we can change it).

To be honest, I don't have more rationale than the one in this message, and in my message above. If this is not convincing enough, let's remove the file. I don't have other arguments.

**Comment by valeriocos** [comment]
which action should be taken to close this issue @jgbarah @jsmanrique ? thanks

**Comment by GeorgLink** [comment]
I vote for 

> remove any content in any identities.yml file in this repository

Because:
1. it still gives an example of how to load an identities.yml file
2. no personal information in the repo files

_optional:_ If the maintainers of this repo want to include their own personal information in the identities.yml file, then that's up to them, but I would not include anyone else.

**Comment by jgbarah** [comment]
I think i have nothing else to say, in addition to my previous comments ;-)
